static/js/components/dashboard/CourseDescription_test.js

```diff
@@ -31,80 +35,102 @@ export function findCourse(courseSelector: (course: Course, program: Program) =>
+    course.runs = [
+      createCourseRun(STATUS_NOT_OFFERED, '2016-02-09T10:20:10Z', '2016-05-01T10:20:10Z')
+    ];
+    const wrapper = shallow(<CourseDescription course={course} />);
+    let firstRun = course.runs[0];
+
+    assert.equal(wrapper.find(".course-description-result").text(), `Coming ${firstRun.fuzzy_start_date}`);
   });
 
-  it('shows Begins ... for verified courses when course start date is in future', () => {
-    let course = findCourse(course => course.status === STATUS_VERIFIED_NOT_COMPLETED);
-    let courseStart = moment(course.runs[0].course_start_date);
-    let yesterday = moment(courseStart).add(-1, 'days');
-    const wrapper = shallow(<CourseDescription course={course} now={yesterday}/>);
-    let formattedDate = courseStart.format(DASHBOARD_FORMAT);
-    assert.equal(wrapper.find(".course-description-result").text(), `Begins ${formattedDate}`);
+  it(`does show date with status verified-not-completed`, () => {
+    let course = _.cloneDeep(findCourse(course => course.status === STATUS_VERIFIED_NOT_COMPLETED));
+    course.runs = [
+      createCourseRun(STATUS_VERIFIED_NOT_COMPLETED, '2016-03-09T10:20:10Z', '2016-06-01T10:20:10Z')
+    ];
+    const wrapper = shallow(<CourseDescription course={course} />);
+    let firstRun = course.runs[0];
+    let courseStartDate = moment(firstRun.course_start_date);
+    let formattedDate = courseStartDate.format(DASHBOARD_FORMAT);
+
+    assert.equal(wrapper.find(".course-description-result").text(), `Start date: ${formattedDate}`);
   });
 
-  it('shows nothing for verified courses when course start date is today or passed', () => {
-    let course = findCourse(course => course.status === STATUS_VERIFIED_NOT_COMPLETED);
-    let courseStart = moment(course.runs[0].course_start_date);
-    let tomorrow = moment(courseStart).add(1, 'days');
+  it(`does show date with status enrolled-not-verified`, () => {
+    let course = _.cloneDeep(findCourse(course => course.status === STATUS_ENROLLED_NOT_VERIFIED));
+    course.runs = [
+      createCourseRun(STATUS_ENROLLED_NOT_VERIFIED, '2016-04-09T10:20:10Z', '2016-07-01T10:20:10Z')
+    ];
+    const wrapper = shallow(<CourseDescription course={course} />);
+    let firstRun = course.runs[0];
+    let courseStartDate = moment(firstRun.course_start_date);
+    let formattedDate = courseStartDate.format(DASHBOARD_FORMAT);
+
+    assert.equal(wrapper.find(".course-description-result").text(), `Start date: ${formattedDate}`);
+  });
 
-    // course starts today
-    let wrapper = shallow(<CourseDescription course={course} now={courseStart}/>);
-    assert.equal(wrapper.find(".course-description-result").text(), "");
+  it(`does show date with status offered-not-enrolled`, () => {
+    let course = _.cloneDeep(findCourse(course => course.status === STATUS_OFFERED_NOT_ENROLLED));
+    course.runs = [
+      createCourseRun(STATUS_OFFERED_NOT_ENROLLED, '2016-05-09T10:20:10Z', '2016-08-01T10:20:10Z')
```

Comment:
i see that you changed the date arguments you're passing into `createCourseRun` for all of these test cases. are these different values important for their test cases? it doesn't look like they are. my earlier suggestion was to set a default value for `courseStartDate` and `courseEndDate` in `createCourseRun`. that way, if the test case doesn't need some specific values for those dates, you can just call `createCourseRun(SOME_STATUS)` and not worry about passing in those values. `createCourseRun` would fill them in if they are undefined